parser/prism/Translator.cc

```diff
@@ -1674,14 +1676,22 @@ unique_ptr<parser::Node> Translator::translateConst(PrismLhsNode *node) {
     auto location = translateLoc(node->base.location);
     auto name = parser.resolveConstant(node->name);
 
-    if (isInMethodDef) { // Check if this is a dynamic constant assignment (SyntaxError at runtime)
-        // This is a copy of a workaround from `Desugar.cc`, which substitues in a fake assignment,
-        // so the parsing can continue. See other usages of `dynamicConstAssign` for more details.
+    if (!skipDynamicConstantWorkaround) {
```

Comment:
Okay... I am not happy with this and would love some thoughts on how to proceed.

I have discovered that it is not illegal to do dynamic constant assignment inside a method IF... you're doing a constant path operator assignment. 🤦🏻 So we can skip the dynamic constant workaround in those cases. However, it is not straightforward to do so because the code path looks like this:

1. `PM_CONSTANT_PATH_OPERATOR_WRITE_NODE` case inside the `translate` method
2. Calls `translateOpAssignment`
3. Calls `translateConst` template and passes the node type as a `pm_constant_path_node`
4. The dynamic constant assignment workaround gets applied

The issue is that when we have a `PM_CONSTANT_PATH_WRITE_NODE` (which should be translated with the dynamic constant workaround), it also gets passed to `translateConst` with the node type as `pm_constant_path_node`. So we need some way to differentiate between these two cases.

Let me know what you think @amomchilov and @st0012.